mathics/builtin/numbers/calculus.py

```diff
@@ -1377,7 +1376,7 @@ class FindRoot(Builtin):
     def apply(self, f, x, x0, evaluation, options):
         "FindRoot[f_, {x_, x0_}, OptionsPattern[]]"
         # First, determine x0 and x
-        x0 = Expression(SymbolN, x0).evaluate(evaluation)
+        x0 = _numeric_evaluation_with_prec(x0, evaluation)
```

Comment:
@mmatera I would appreciate it if we could try to regularize the names - whether or not this had been done previously. 

Functions that are basically concrete `apply()` method  of Builtin Class X should follow a distinct pattern. In the future we may autogenerate this.

Reply:
The point here is that there are certain functions that I think shouldn't be implemented as member functions, to avoid using the Mathics mechanism of calling functions recursively. This is one case:
`_numeric_evaluation_with_prec(x0, evaluation, prec)` 
should do the same that `Expression("N",x0, evaluation, prec)`, but internally, without all the recursive pattern matching, which is very slow.

On the other hand, if someone has a better name for this function, let's change it.

Reply:
> The point here is that there are certain functions that I think shouldn't be implemented as member functions, to avoid using the Mathics mechanism of calling functions recursively.

There wasn't any recursive pattern matching using `evaluate_fast()` and the `sapply..`

 This is one case:
> `_numeric_evaluation_with_prec(x0, evaluation, prec)`
> should do the same that `Expression("N",x0, evaluation, prec)`, but internally, without all the recursive pattern matching, which is very slow.

The difference it seems between this and `sapply` is simply that one is a class method and the other isn't in a class at all. In terms of the difference in time there is some, but it is very small. And that's not at all where the time is going.

Reply:
OK, but what do you think is the best? staticmember methods or plane Python functions? I was not sure and for this reason I use one of then on each case...

Reply:
I have to think about it, and try some experiments and maybe benchmark. So lemme get back to you on this. In some cases though when a function uses class members like constant

Reply:
@mmatera I guess it can be used [here](https://github.com/mathics/Mathics/blob/faster_quick_pattern_test/mathics/builtin/arithmetic.py#L137) too.